Structure_Engine/Query/AnalyticalBars.cs

```diff
@@ -101,7 +101,12 @@ private static List<Bar> AnalyticalBars(BHP.FramingProperties.ConstantFramingPro
 
             ISectionProperty section;
 
-            if (convertedProps.ContainsKey(property))
+            if (property == null)
+            {
+                Engine.Base.Compute.RecordError("One or more properties are null. Please make sure all of the inputs have a value.");
+                return new List<Bar>();
+            }
+            else if (convertedProps.ContainsKey(property))
```

Comment:
Is this supposed to not return anything if a property is not given? If so I am fine to approve here. (as far as I can tell, after this, the method seems to expect a section derived from this property, Just asking for confirmation really)